File: php/features.php
<?php
declare(strict_types=1);

// LmSms SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class LmSmsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new LmSmsBaseFeature();
            case "test":
                return new LmSmsTestFeature();
            default:
                return new LmSmsBaseFeature();
        }
    }
}

File: py/features.py
# LmSms SDK feature factory

from feature.base_feature import LmSmsBaseFeature
from feature.test_feature import LmSmsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: LmSmsBaseFeature(),
        "test": lambda: LmSmsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
